### ISSUE TITLE
az vm create should let Azure decide on what the default storage account type is when using managed disks

### ISSUE THREAD

**Issue by johanste** [Azure/azure-cli#4129]
Today, the CLI defaults to Premium_LRS. Unless explicitly specified on the comand line, we shouldn't have to set this since the service will automatically choose for us.

**Comment by johanste** [comment]
Not sure why this was labeled as a breaking change. Removed label since it isn't - or am I missing something?

**Comment by lmazuel** [comment]
+1

**Comment by tjprescott** [comment]
If you had a script that used the previous default, now that could be different and thus the script would behave differently. That is a breaking change.

**Comment by lmazuel** [comment]
@tjprescott depends, what is the default value of managed disk RestAPI right now? If it's Premium_LRS, it's not breaking. And since a provider is not supposed to change a default value for a given released ApiVersion, this should be transparent to user.
Otherwise, If current RestAPI defaults to something else, I would even say it's a CLI bug, I don't think it's SDK/CLI responsibility to decide what it a good default for Azure if the RestAPI is able to provide one.
What will you respond to customer that would open an issue saying "CLI default value is not the same that RestAPI default value". It's difficult to argue that it's by design and not a bug :/

**Comment by johanste** [comment]
The current version is using Premium_LRS if not specified. If the VM size chosen does not support premium storage, the command would fail. If we do take this change, and you do specify the storage SKU, then I would assume that we honor whatever was specified on the command line and make an attempt (else you wouldn't be able to use non-premium storage for a VM size that supported premium storage).

So, if we consider a switch from "the command previously failed with the given input, but now it works" as a breaking change, then it would be breaking... I don't think our bar for change in behavior is quite that strict :)

**Comment by tjprescott** [comment]
In that case of course it wouldn't be, but in the case where Premium IS supported but isn't the default it would be. In that case, the current CLI would default to Premium, while the service would default to Standard. That is the breaking scenario I'm thinking of.

**Comment by johanste** [comment]
Yeah, that scenario doesn't exist today to the best of my knowledge.
